### DISCUSSION TITLE
set_claim returns "access denied" when called from a trigger function, even with SECURITY DEFINER enabled

### DISCUSSION THREAD

**Discussion by Vittorio94** [supabase/supabase#13092] [Questions]
Hello everyone,

I am trying to implement custom claims. I have a function that gets triggered when a new row is inserted or updated inside a table called "transaction". The function has the SECURITY DEFINER parameter enabled. Inside this function I call set_claim to update the custom claims of a user (user_id is one of the columns of the table) based on the values of the new row.

When I insert a new row from the dashboard or from psql, everything works fine and the custom claims are updated correctly (set_claim returns "OK").

When I do it from the javascript library, set_claim returns an "access denied" error. This even happens when I use the secret api key instead of the public one. 

Anybody has ideas on how to solve this?

Thank you

**Comment by tiagofsanchez** [comment]
Hi @Vittorio94, 

I am having the same problem as well. On my dashboard I am able to do everything, however, I am not able to do anything on the client side.

Example of my function on the client side: 
```
async function checkProfileClaims() {
    try {
      const updates = {
        uid: userData.id,
        // claim: "userrole",
        // value: "Broker",
      };
      let { error, data } = await supabase?.rpc('get_claims', {...updates});
      if (error) throw error;
      console.log({ data, updates });
    } catch (error) {
      alert(JSON.stringify(error, null, 2));
      console.log({ error, updates });
    }
  }
```

Where the console.log() outputs the following: 
```
data: {error: 'access denied'}
updates: {uid: 'a31693ea-6a5a-4e0b-ad89-53baaf2eda97'
```

Any luck with this?
